Linux C libs are moving away from implicit header pollution with sys/types.h

--- a/include/qemu/osdep.h
+++ b/include/qemu/osdep.h
@@ -78,6 +78,10 @@ extern int daemon(int, int);
 #include <assert.h>
 #include <signal.h>
 
+#ifdef __linux__
+#include <sys/sysmacros.h>
+#endif
+
 #ifdef __OpenBSD__
 #include <sys/signal.h>
 #endif
